themes/classic/templates/catalog/_partials/product-details.tpl

```diff
@@ -30,7 +30,8 @@
     {if $product.show_quantities}
       <div class="product-quantities">
         <label class="label">{l s='In stock' d='Shop.Theme.Catalog'}</label>
-        <span>{$product.quantity} {$product.quantity_label}</span>
+        <span data-stock="{$product.quantity}">{$product.quantity} {$product.quantity_label}</span>
+        <span data-allow-oosp="{$product.allow_oosp}"></span>
```

Comment:
You can add more than one data-attribute to an element.

I don't see any reason to keep this new `<span>`, so please merge it with the one on line 33.